src/test/shell/bazel/bazel_coverage_test.sh

```diff
@@ -66,29 +62,143 @@ int main(void) {
   a(true);
 }
 EOF
+}
 
-  bazel coverage --test_output=all --build_event_text_file=bep.txt //:t \
-      &>$TEST_log || fail "Coverage for //:t failed"
+# Returns the expected coverage result in lcov format for a given source file.
+#
+# - source_file The name of the source file for which the expected coverage
+#               should be returned. The accepted values are "a.cc" and "t.cc".
+#               For other values the method fails.
+function get_expected_coverage() {
+    local source_file="${1}"
+
+    # The expected results can be constructed manually by following the lcov
+    # documentation and checking what lines of code are covered when
+    # running the test.
+    # For details about the lcov format see
+    # http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
+
+    # The expected coverage result for a.cc in lcov format.
+    local expected_result_a_cc="TN:
+SF:a.cc
+FN:3,_Z1ab
+FNDA:1,_Z1ab
+FNF:1
+FNH:1
+DA:3,1
+DA:4,1
+DA:5,1
+DA:7,0
+LF:4
+LH:3
+end_of_record"
+
+    # The expected coverage result for t.cc in lcov format.
+    local expected_result_t_cc="TN:
+SF:t.cc
+FN:4,main
+FNDA:1,main
+FNF:1
+FNH:1
+DA:4,1
+DA:5,1
+DA:6,1
+LF:3
+LH:3
+end_of_record"
+
+    # Return the expected result for the given source file.
+    case "$source_file" in
+        ("a.cc") echo "$expected_lcov_result_a_cc" ;;
+        ("t.cc") echo "$expected_lcov_result_t_cc" ;;
+        (*) fail "Coverage cannot be requested for file $source_file" ;;
+    esac
+}
+
+# Asserts if the given expected coverage result is included in the given output
+# file.
+#
+# - expected_coverage The expected result that must be included in the output.
+# - output_file       The location of the coverage output file.
+function assert_coverage_result() {
+    local expected_coverage="${1}"
+    local output_file="${2}"
+
+    # Replace newlines with commas to facilitate the assertion.
+    local expected_coverage_no_newlines=$( echo "$expected_coverage" | tr '\n' ',' )
+    local output_file_no_newlines=$( cat "$output_file" | tr '\n' ',' )
+
+    ( echo $output_file_no_newlines | grep  $expected_coverage_no_newlines ) \
+        || fail "Expected coverage result
+<$expected_coverage>
+was not found in actual coverage report:
+<$( cat $output_file )>"
+}
 
-  ending_part=$(sed -n -e '/PASSED/,$p' $TEST_log)
+# Returns the path of the code coverage report that was generated by Bazel by
+# looking at the current $TEST_log. The method fails if TEST_log does not
+# contain any coverage report for a passed test.
+function get_coverage_file() {
+  local ending_part=$(sed -n -e '/PASSED/,$p' $TEST_log)
 
-  coverage_file_path=$(grep -Eo "/[/a-zA-Z0-9\.\_\-]+\.dat$" <<< "$ending_part")
+  local coverage_file_path=$(grep -Eo "/[/a-zA-Z0-9\.\_\-]+\.dat$" <<< "$ending_part")
   [ -e $coverage_file_path ] || fail "Coverage output file does not exist!"
```

Comment:
I know you're not modifying this line but if you can use double brackets [[, https://google.github.io/styleguide/shell.xml?showone=Test,_%5B_and_%5B%5B#Test,_%5B_and_%5B%5B